src/Orleans.Transactions/InClusterTM/TransactionManager.cs

```diff
@@ -5,10 +5,99 @@
 using System.Threading;
 using Microsoft.Extensions.Options;
 using Microsoft.Extensions.Logging;
+using Orleans.Runtime;
 using Orleans.Transactions.Abstractions;
+using Orleans.Runtime.Configuration;
 
 namespace Orleans.Transactions
 {
+    internal class TransactionManagerMetrics
+    {
+        private const string StartTransactionRequestTPS = "TransactionManager.StartTransaction.TPS";
+        private const string AbortTransactionRequestTPS = "TransactionManager.AbortTransaction.TPS";
+        private const string CommitTransactionRequestTPS = "TransactionManager.CommitTransaction.TPS";
+       
+        private const string AbortedTransactionDueToDependencyTPS = "Transaction.AbortedDueToDependency.TPS";
+
+        private const string AbortedTransactionDueToMissingInfoInTransactionTableTPS =
+            "Transaction.AbortedDueToMissingInfoInTransactionTable.TPS";
+        private const string AbortedTransactionTPS = "Transaction.Aborted.TPS";
+        //pending transaction produced TPS in the monitor window
+        private const string PendingTransactionTPS = "Transaction.Pending.TPS";
+        //TPS for validated transaction in the monitor window
+        private const string ValidatedTransactionTPS = "Transaction.Validated.TPS";
+
+        internal long StartTransactionRequestCounter { get; set; }
+        internal long AbortTransactionRequestCounter { get; set; }
+        internal long CommitTransactionRequestCounter { get; set; }
+        internal long AbortedTransactionCounter { get; set; }
+        internal long AbortedTransactionDueToDependencyCounter { get; set; }
+        internal long AbortedTransactionDueToMissingInfoInTransactionTableCounter { get; set; }
+        internal long PendingTransactionCounter {get;set;}
+        internal long ValidatedTransactionCounter { get; set; }
+
+        private DateTime lastReportTime = DateTime.Now;
+        private ITelemetryProducer telemetryProducer;
+        private PeriodicAction monitor;
+        public TransactionManagerMetrics(ITelemetryProducer telemetryProducer, TimeSpan metricReportInterval)
+        {
+            this.telemetryProducer = telemetryProducer;
+            this.monitor = new PeriodicAction(metricReportInterval, this.ReportMetrics);
+        }
+
+        public void TryReportMetrics()
+        {
+            this.monitor.TryAction(DateTime.Now);
+        }
+
+        private void ResetCounters(DateTime lastReportTime)
+        {
+            this.lastReportTime = lastReportTime;
+            this.StartTransactionRequestCounter = 0;
+            this.AbortTransactionRequestCounter = 0;
+            this.CommitTransactionRequestCounter = 0;
```

Comment:
I don't quite understand why these 3 vars aren't past tense while the vars below are past tense.

Reply:
Because for example, CommitTransactionRequest is a word, it is the counter for requests.

AbortedTransaction is a word, it is the counter for transaction which are aborted. Does this make sense?

Reply:
Isn't `AbortTransactionRequestCounter` always the same as `AbortedTransactionCounter`, since they're always at the same time?

Reply:
Yeah, they are counting the same thing. It is clear to you because you read the code. But think in the aspect of people who use this metric. They would categorize them into two category, request counter related metric and transaction status related metrics. It may not be clear to them in the first sight, without reading the code, that AbortedTransactionRequestCOunter, and AbortedTransactionCounter is the same thing. 

I maybe over-thinking on this, but just share my thought on this.